app/scripts/controllers/detect-tokens.js

```diff
@@ -48,7 +51,9 @@ export default class DetectTokensController {
       if (
         contracts[contractAddress].erc20 &&
         !this.tokenAddresses.includes(contractAddress.toLowerCase()) &&
-        !this.hiddenTokens.includes(contractAddress.toLowerCase())
+        !this.hiddenTokens.find((token) =>
+          isEqualCaseInsensitive(token.address, contractAddress),
```

Comment:
@NiranjanaBinoy see this `isEqualCaseInsensitive` method, as an alternative to how you handle case differences in your PR. I think this is okay for comparison of token addresses. You will have to change to this within this file, once you rebase.